src/python_testing/TCP_Tests.py

```diff
@@ -0,0 +1,150 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs: run1
+# test-runner-run/run1/app: ${ALL_CLUSTERS_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+# === END CI TEST ARGUMENTS ===
+#
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TCP_Tests(MatterBaseTest):
+
+    @async_test_body
+    async def test_TCPConnectionEstablishment(self):
+
+        try:
+            device = await self.default_controller.GetConnectedDevice(nodeid=self.dut_node_id, allowPASE=False, timeoutMs=1000,
+                                                                      payloadCapability=ChipDeviceCtrl.TransportPayloadCapability.LARGE_PAYLOAD)
+        except TimeoutError:
+            asserts.fail("Unable to establish a CASE session over TCP to the device")
```

Comment:
I think exceptions automatically fail tests. Is there value to convert a exception trace (which is long/stack so somewhat ugly) with a nice failure text?

I am comparing this with the downside of having try/catch blocks and indents (i.e. uglier code to maintain vs uglier error messages to view on failures)

Reply:
While the stack trace is ugly, it does provide the method call chain(which may be useful in some contexts, but not all). As for the additional failure text, it's purpose is probably to serve more context that the trace cannot provide. 
For the tests, I have adopted some of the existing patterns of catching IM exceptions on IM operations and Timeout exceptions for connection establishments.